utils/src/main/kotlin/OrtAuthenticator.kt

```diff
@@ -86,7 +86,7 @@ class OrtAuthenticator(private val original: Authenticator? = null) : Authentica
             RequestorType.PROXY -> {
                 val proxySelector = ProxySelector.getDefault()
                 if (proxySelector is OrtProxySelector) {
-                    val type = requestingProtocol.toProxyType() ?: return null
+                    val type = requestingProtocol.toProxyType() ?: return super.getPasswordAuthentication()
```

Comment:
Why? The super implementation also always returns `null`.

Reply:
For consistency with the final return statement where it is already done that way. Also, I do consider it to be good practice to not hard-code the value here.

Reply:
The commit message makes it sound like there was a chance that the super class does not return `null`.

Reply:
I've reworded the commit message and dropped the last two commits, as the issue I was having with NTLM authentication first needs to be debugged further.